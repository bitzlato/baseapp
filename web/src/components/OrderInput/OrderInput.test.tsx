import { shallow } from 'enzyme';
import { OrderInput, OrderInputProps } from '.';

const defaultProps: OrderInputProps = {
  currency: 'eth',
  handleChangeValue: () => undefined,
  handleFocusInput: () => undefined,
  value: '',
  onKeyPress: jest.fn(),
};

const setup = (props: Partial<OrderInputProps> = {}) =>
  shallow(<OrderInput {...{ ...defaultProps, ...props }} />);

describe('InputBlock', () => {
  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should have correct className', () => {
    const wrapper = setup();
    expect(wrapper.hasClass('cr-order-input')).toBeTruthy();
  });

  it('should pass along supplied className', () => {
    const className = 'some-class';
    const wrapper = setup({ className });
    expect(wrapper.hasClass(className)).toBeTruthy();
  });
});
