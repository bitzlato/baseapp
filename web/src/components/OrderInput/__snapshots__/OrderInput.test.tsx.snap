// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBlock should render 1`] = `
<div
  className="cr-order-input"
>
  <CustomInput
    defaultLabel=""
    handleChangeInput={[Function]}
    handleFocusInput={[Function]}
    inputValue=""
    label=""
    onKeyPress={[MockFunction]}
    placeholder="0"
    type="number"
  />
  <CurrencyTicker
    className="cr-order-input__currency"
    symbol="eth"
  />
</div>
`;
